components/core/src/clp/streaming_compression/lzma/Decompressor.hpp

```diff
@@ -0,0 +1,105 @@
+#ifndef CLP_STREAMING_COMPRESSION_LZMA_DECOMPRESSOR_HPP
+#define CLP_STREAMING_COMPRESSION_LZMA_DECOMPRESSOR_HPP
+
+#include <cstddef>
+
+#include "../../ErrorCode.hpp"
+#include "../../ReaderInterface.hpp"
+#include "../../TraceableException.hpp"
+#include "../Constants.hpp"
+#include "../Decompressor.hpp"
+
+namespace clp::streaming_compression::lzma {
+class Decompressor : public ::clp::streaming_compression::Decompressor {
+public:
+    // Types
+    class OperationFailed : public TraceableException {
+    public:
+        // Constructors
+        OperationFailed(ErrorCode error_code, char const* const filename, int line_number)
+                : TraceableException(error_code, filename, line_number) {}
+
+        // Methods
+        [[nodiscard]] auto what() const noexcept -> char const* override {
+            return "streaming_compression::lzma::Decompressor operation failed";
+        }
+    };
+
+    // Constructor
+    Decompressor() : clp::streaming_compression::Decompressor(CompressorType::LZMA) {}
+
+    // Destructor
+    ~Decompressor() override = default;
+
+    // Delete copy constructor and assignment operator
+    Decompressor(Decompressor const&) = delete;
+    auto operator=(Decompressor const&) -> Decompressor& = delete;
+
+    // Delete move constructor and assignment operator
+    // TODO: change to default when the base decompressor class has been updated
+    Decompressor(Decompressor&&) noexcept = delete;
+    auto operator=(Decompressor&&) noexcept -> Decompressor& = delete;
+
+    // Methods implementing the ReaderInterface
+    /**
+     * Tries to read up to a given number of bytes from the decompressor.
+     * @param buf
+     * @param num_bytes_to_read The number of bytes to try reading
+     * @param num_bytes_read The actual number of bytes read
+     * @return ErrorCode_Unsupported
+     */
+    [[nodiscard]] auto try_read(char* buf, size_t num_bytes_to_read, size_t& num_bytes_read)
+            -> ErrorCode override;
+
+    /**
+     * Tries to seek from the beginning to the given position.
+     * @param pos
+     * @return ErrorCode_Unsupported
+     */
+    [[nodiscard]] auto try_seek_from_begin(size_t pos) -> ErrorCode override;
+
+    /**
+     * Tries to get the current position of the read head.
+     * @param pos Position of the read head in the file
```

Comment:
```suggestion
     * @param pos Returns the position of the read head in the buffer.
```
We should use `Returns ...` for an output parameter